src/docs/webassembly-opcode.md

```diff
@@ -0,0 +1,430 @@
+     // clang-format off
+
+     // Standard opcodes
++    CASE_I32_OP(Add1, "add1")
+     CASE_INT_OP(Eqz, "eqz")
+     CASE_ALL_OP(Eq, "eq")
+     CASE_I64x2_OP(Eq, "eq")
+```
+
+By adding our new instruction in `FOREACH_SIMPLE_OPCODE`, we are skipping a [fair amount of work](https://cs.chromium.org/chromium/src/v8/src/wasm/function-body-decoder-impl.h?l=1751-1756&rcl=686b68edf9f42c201c2b25bca9f4bef72ff41c0b) that is done in `src/wasm/function-body-decoder-impl.h`, which decodes Wasm opcodes and calls into the Turbofan graph generator. Thus, depending on what your opcode does, you might have more work to do. We skip this in the interest of brevity.
+
+## Writing a test for the new opcode
+
+Wasm tests can be found in [`test/cctest/wasm/`](https://cs.chromium.org/chromium/src/v8/test/cctest/wasm/).  We will look at [`test/cctest/wasm/test-run-wasm.cc`](https://cs.chromium.org/chromium/src/v8/test/cctest/wasm/test-run-wasm.cc), where many "simple" opcodes are tested.
+
+There are many examples in this file that we can follow. The general setup is:
+
+- create a `WasmRunner`
+- set up globals to hold result (optional)
+- set up locals as parameters to instruction (optional)
+- build the wasm module
+- run it and compare with an expected output
+
+Here's a simple test for our new opcode:
+
+```diff
+diff --git a/test/cctest/wasm/test-run-wasm.cc b/test/cctest/wasm/test-run-wasm.cc
+index 26df61ceb8..b1ee6edd71 100644
+--- a/test/cctest/wasm/test-run-wasm.cc
++++ b/test/cctest/wasm/test-run-wasm.cc
+@@ -28,6 +28,15 @@ namespace test_run_wasm {
+ #define RET(x) x, kExprReturn
+ #define RET_I8(x) WASM_I32V_2(x), kExprReturn
+
++#define WASM_I32_ADD1(x) x, kExprI32Add1
++
++WASM_EXEC_TEST(Int32Add1) {
++  WasmRunner<int32_t> r(execution_tier);
++  // 10 + 1
++  BUILD(r, WASM_I32_ADD1(WASM_I32V_1(10)));
++  CHECK_EQ(11, r.Call());
++}
++
+ WASM_EXEC_TEST(Int32Const) {
+   WasmRunner<int32_t> r(execution_tier);
+   const int32_t kExpectedValue = 0x11223344;
+```
+
+Run the test:
+
+```bash
+$ tools/dev/gm.py x64.debug 'cctest/test-run-wasm-simd/RunWasmTurbofan_I32Add1'
+...
+=== cctest/test-run-wasm/RunWasmTurbofan_Int32Add1 ===
+#
+# Fatal error in ../../src/compiler/wasm-compiler.cc, line 988
+# Unsupported opcode 0xee:i32.add1
+```
+
+Tip: finding the test name can be tricky, since the test definition is behind a macro. Use [Code Search](https://cs.chromium.org) to click around to discover the macro definitions.
+
+This error indicates that the compiler does not know of our new instruction - that will change in the next section.
+
+## Compiling Wasm into Turbofan
+
+In the introduction, we mentioned that Wasm instructions are compiled into a Turbofan graph, `wasm-compiler.cc` is the where this happens. Let's take a look at an example opcode, [I32Eqz](https://cs.chromium.org/chromium/src/v8/src/compiler/wasm-compiler.cc?l=716&rcl=686b68edf9f42c201c2b25bca9f4bef72ff41c0b):
+
+```
```

Comment:
Probably worth adding a language to the block? (not 100% sure how this Markdown is rendered, but usually you need it for the syntax highlight to work)